onnxruntime/core/providers/cpu/nn/batch_norm.cc

```diff
@@ -37,11 +37,16 @@ ONNX_CPU_OPERATOR_KERNEL(
     BatchNormalization,
     9,
     KernelDefBuilder()
-        .TypeConstraint("X", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("scale", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("B", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("mean", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("var", DataTypeImpl::GetTensorType<float>()),
+        .TypeConstraint("X", {DataTypeImpl::GetTensorType<float>(),
```

Comment:
So X could be double while scale is float?

Reply:
I have set T to float or double, could you take a look now?